src/frontend/src/views/Data/index.tsx

```diff
@@ -25,11 +25,22 @@ const TITLE: Record<string, string> = {
   [ROUTES.PHYLO_TREES]: "Phylogenetic Trees",
 };
 
+const mapObjectArrayToIdDict = (
```

Comment:
I know the intent of this func is to map to ID, but it's not actually coded to inherently do that. It's generalized, which is probably a good thing, but `keyString` could be anything: it could be run on `uploadDate` or whatever, right? I think either the name of the func should change to indicate what it really does (although I'm struggling with a good, not insanely verbose name... it's like a thing that "reduces an array of objects to a mapping between the keyString arg and the objects that make up the array" ... hmmm, maybe `reduceObjectArrayToLookupDict`? then have the keyString arg become `keyStringForLookups` or something?) OR make it less generalized and have the `keyString` not be an arg, but rather make it `dataType` or something which is one of "sample" or "tree" or whatever, then the func conditionally chooses the `keyString` based on if it's on the sample track or the tree track.

If you do decide to keep it general, I think a one line comment explaining that it's converting an array of objects into a lookup of those same objects based on the provided key string would be helpful, it definitely took me a couple reads to figure out that was what I was seeing.